torch/csrc/cuda/expand_utils.cpp

```diff
@@ -0,0 +1,188 @@
+#include "torch/csrc/cuda/THCP.h"
```

Comment:
This is tricky, and I'll add a comment, although it's already done in a number of places in the code.  The functions in expand_utils.h/expand_utils-inl.h are defined in terms of the LIBRARY_STATE and related macros (because the copy functions are defined the same way).  So, there are actually two versions of each function (with and without THCState) and we avoid redefining the non-cuda ones by including THCP.h (that in terms defines the macros) and then including those files.